core/consensus/component.go

```diff
@@ -289,3 +296,18 @@ func (c *Component) getPeerIdx() (int64, error) {
 func isContextErr(err error) bool {
 	return errors.Is(err, context.DeadlineExceeded) || errors.Is(err, context.Canceled)
 }
+
+type spanKey struct{}
```

Comment:
why empty struct?

Reply:
See `context.WithValue` godoc:
```
// The provided key must be comparable and should not be of type
// string or any other built-in type to avoid collisions between
// packages using context. Users of WithValue should define their own
// types for keys. To avoid allocating when assigning to an
// interface{}, context keys often have concrete type
// struct{}. Alternatively, exported context key variables' static
// type should be a pointer or interface.
```

Reply:
cool